### DISCUSSION TITLE
What is `shared state`?

### DISCUSSION THREAD

**Discussion by hyunbinseo** [sveltejs/kit#4343] [Q&A] [answered]
- What is `shared state` in the following context?
- How is `shared state` mutated on the server?

> It is recommended that you not store pre-request state in global variables, but instead use them only for cross-cutting concerns such as caching and holding database connections.

> Mutating any shared state on the server will affect all clients, not just the current one.

Source: [Loading, SvelteKit Documentation](https://kit.svelte.dev/docs/loading)

---

This discussion will answer https://github.com/sveltejs/kit/issues/2213#issuecomment-910497495.

**Comment by benmccann** [comment, marked as answer]
state: the particular condition something is in at a specific time
shared: possessed in common with another or others

"shared state" in this context essentially means a variable shared by two different requests. Using `svelte/store` wouldn't be an example of shared state if you created a new instance of a store for each request, but often people will create a store as a global variable which causes it to be shared across requests

**Reply by hyunbinseo** [reply]
I have been importing `store` from `.ts` files as mentioned in https://github.com/sveltejs/kit/discussions/4339#discussioncomment-2374963 and https://github.com/sveltejs/kit/discussions/4339#discussioncomment-2377561.

1. Would this impact static sites generated with `@sveltejs/adapter-static`?
2. Would this impact serverless deployments, such as `Netlify` or `Cloudflare Workers`?

I hope this use-case is implicitly mentioned in the documentation as mentioned in https://github.com/sveltejs/kit/discussions/4339#discussioncomment-2374935.

> If your store lives in a global variable, it's a global variable and therefore is subject to the same semantics. Maybe we should also explicitly mention that this also applies to store just to make this clear.

**Reply by benmccann** [reply]
There is no code that runs on the server for a static site, so it would not apply there. Serverless sites work the same as traditional servers, so it would apply to serverless

**Comment by dominikg** [comment]
See also https://github.com/sveltejs/kit/discussions/4339

**Comment by hyunbinseo** [comment]
See also: [What is Singleton?](https://youtu.be/xhi0m1QZue0) by [Tan Li Hau](https://github.com/tanhauhau)
